pkg/metrics/experimental_register.go

```diff
@@ -0,0 +1,40 @@
+// +build experimental_metrics
+
+package metrics
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	// Register experimental metrics
+	reconcileMetrics = reconcileCounters(operatorController, adoptionCSVController, adoptionSubscriptionController, operatorConditionController, operatorConditionGeneratorController)
+	registerReconcileMetrics()
+}
+
+func reconcileCounters(reconcilerNames ...string) map[string]prometheus.CounterVec {
+	result := map[string]prometheus.Counter{}
+	for _, s := range reconcilerNames {
+		result[s] = reconcileCounter(s)
+	}
+	return result
+}
+
+func reconcileCounter(name string) prometheus.CounterVec {
+	return prometheus.NewCounterVec(
```

Comment:
Poking at the godoc for the Prometheus library, it looks like the signature returns a pointer to a prometheus.CounterVec, but we're expecting to return the value of that variable, vs. the reference in reconcileCounter.

Reply:
Thanks for your patience @timflannagan, I am working on getting this over the finish line now. I am not sure that I am capturing what you're trying to say here.

Reply:
It looks like you've addressed this issue since the last rebase + updates. This was an issue where we were using the wrong type definition, so we had seen that compilation was failing throughout the various CI checks.